### PR TITLE
1403 add strive molecular data overlays

### PR DESCRIPTION
**Pull request by avaek**
Addresses: https://github.com/beyondessential/tupaia-backlog/issues/1403

### CODE REVIEW

File: packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    measureLevel: 'Facility',
+    hideFromLegend: true,
+  },
+  countryCodes: '{PG}',
+  projectCodes: '{strive,explore}',
+};
+
+const STR_QMAL05_OVERLAY = {
+  ...BASE_OVERLAY,
+  id: generateId(),
```
Comment (Alastair-L) [inline packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js:46]: Generally the map overlays have more expressive ids than this. Your last one was good (`PG_STRIVE_K13_C580Y_Positive`)

File: packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    ],
+    displayType: 'radius',
+    measureLevel: 'Facility',
+    hideFromLegend: true,
+  },
+  countryCodes: '{PG}',
+  projectCodes: '{strive,explore}',
+};
+
+const STR_QMAL05_OVERLAY = {
```
Comment (Alastair-L) [inline packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js:44]: a) I like the way that you've reduced code duplication with BASE_OVERLAY

b) I know it's nitpicky, so feel free to ignore me here, but there's still a bit of duplication that could be removed here. A pattern that could work here (or next time) is 
```javascript
OVERLAYS_TO_IMPORT = [
   {
      id: 'something',
      dataElementCode: 'abc',
      name: 'boo!',
      programCode: 'hi',
   },
   //...
]
```
And then in the up migration:
```javascript
OVERLAYS_TO_INSERT.forEach(
({ id, name, dataElementCode, programCode }) => {
    overlay = {
      ...BASE_OVERLAY,
      id,
      name,
      measureBuilderConfig: {
        dataValues: {
          [dataElementCode]: 1,
        },
        programCode,
        entityAggregation: {
          aggregationType: 'REPLACE_ORG_UNIT_WITH_ORG_GROUP',
          dataSourceEntityType: 'village',
          aggregationEntityType: 'facility',
        },
      };
    // Import etc
});
```

Just a thought

File: packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  const STRIVE_MOLECULAR_DATA_ID = await db.runSql(`
+     SELECT "id" FROM "map_overlay_group" WHERE "code" = 'STRIVE_Molecular_Data';
+  `);
+
+  OVERLAYS_TO_IMPORT.forEach(overlay => {
+    const MAP_OVERLAY_RELATION_ID = generateId();
+
+    insertObject(db, 'mapOverlay', overlay);
+
+    db.runSql(`
```
Comment (Alastair-L) [inline packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js:147]: Remember `insertObject` works on every table! 🎉

File: packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  });
+};
+
+exports.down = async function (db) {
+  const STRIVE_MOLECULAR_DATA_ID = await db.runSql(`
+    SELECT "id" FROM "map_overlay_group" WHERE "code" = 'STRIVE_Molecular_Data';
+  `);
+
+  OVERLAYS_TO_IMPORT.forEach(async overlay => {
+    const OVERLAY_ID_TO_DELETE = await db.runSql(`
```
Comment (Alastair-L) [inline packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js:160]: This call can be removed with static ids 🎉

**Review by Alastair-L — CHANGES_REQUESTED** [review]
Looks good! Just a few style notes which you can take or leave, and making the id meaningful.

**Comment by avaek** [comment]
Thanks for the great suggestions @Alastair-L - I believe I've implemented them correctly, let me know what you think

File: packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js [hunk 5]
```diff
@@ -41,113 +41,70 @@ const BASE_OVERLAY = {
...
-  ...BASE_OVERLAY,
-  id: generateId(),
-  name: 'QMAL Positive',
-  measureBuilderConfig: {
-    dataValues: {
-      STR_QMAL05: 1,
-    },
+const OVERLAYS_TO_IMPORT = [
+  {
+    id: 'STR_QMAL05_POSITIVE',
```
Comment (Alastair-L) [inline packages/database/src/migrations/20210205033535-AddSTRIVEMolecularDataOverlays-modifies-data.js:46]: This is very nitpicky, but the existing overlay in that group has id: `PG_STRIVE_K13_C580Y_Positive`. Did you have a particular reason to deviate from that pattern?

**Review by Alastair-L — APPROVED** [review]
Looks great! I love seeing all that red :)

**Comment by avaek** [comment]
> This is very nitpicky, but the existing overlay in that group has id: PG_STRIVE_K13_C580Y_Positive. Did you have a particular reason to deviate from that pattern?

Great point! The format I followed was just based on a random map overlay in the STRIVE project but it makes sense to copy the `id` format of the existing map overlays in that particular map overlay group. I've updated the migration accordingly

**Comment by Alastair-L** [comment]
Already approved, but consider it double approved now :)

**Pull request merged**